ambry-messageformat/src/test/java/com/github/ambry/messageformat/BlobStoreHardDeleteTest.java

```diff
@@ -338,18 +338,18 @@ public void blobStoreHardDeleteTestBlobV2Simple() throws MessageFormatException,
     short[] blobVersions = new short[5];
     BlobType[] blobTypes = new BlobType[5];
     for (int i = 0; i < 5; i++) {
-      blobVersions[i] = MessageFormatRecord.Blob_Version_V2;
+      blobVersions[i] = MessageFormatRecord.Blob_Version_V3;
```

Comment:
This test continues to test V1 and V2 de-serialization from stream, but this test does not test V2 serialization to stream.  There are other unit tests that test V2 serialization and de-serialization.

Those tests need to serialize a blob to stream, as part of the test flow.  The serialization should use the latest version only V3, just like all other record formats.  We should not serialize new put operations using an older blob versions.  If we support serializing put operations using old format version, which could lead to data lost, it will make unit testing very complex because every serialize call need to pass in a version.  Imagine the PutRequest needs to pass in header version, blob property version, user metadata version, and so on.  That would be messy and unnecessary.  So let's just use the latest version in the put operation, and support all previous versions in the get operation.